src/firestore/app/run.ts

```diff
@@ -199,6 +200,30 @@ export class RoarRun {
     this.started = true;
   }
 
+  /**
+   * Add engagement flags to a run.
+   * @method
+   * @async
+   * @param {string[]} engagementFlags - Engagement flags to add to the run
+   * @param {boolean} markAsReliable - Whether or not to mark the run as reliable, defaults to false 
+   */
+  async addEngagementFlags(engagementFlags: string[], markAsReliable = false) {
+    if (!this.started) {
+      throw new Error('Run has not been started yet. Use the startRun method first.');
+    }
+
+    const engagementObj = engagementFlags.reduce((acc: { [x: string]: unknown }, flag) => {
+      acc[flag] = true;
+      return acc;
+    }, {});
+
+    if (!this.aborted) {
+      return updateDoc(this.runRef, { engagementFlags: engagementObj, reliable: markAsReliable });
```

Comment:
As it's written right now, this will overwrite any engagement flags that are already there. I think that might be okay but I just want to make sure that it's explicit. Can you add a comment to the docstring of both this method and the appkit method to let the user know that they will be overwriting any existing engagement flags for this run? It will then be the ROAR-app developer's responsibility to keep track of any previously assigned engagement flags.